mobile/src/main/java/ch/epfl/sdp/mobile/ui/ar/ChessScene.kt

```diff
@@ -0,0 +1,123 @@
+package ch.epfl.sdp.mobile.ui.ar
+
+import android.content.Context
+import androidx.lifecycle.LifecycleCoroutineScope
+import ch.epfl.sdp.mobile.application.chess.engine.*
+import ch.epfl.sdp.mobile.ui.*
+import com.google.ar.sceneform.math.Vector3
+import io.github.sceneview.ar.node.ArModelNode
+import io.github.sceneview.ar.node.PlacementMode
+import io.github.sceneview.math.Rotation
+import io.github.sceneview.node.ModelNode
+
+/**
+ * This class represent the AR chess scene which contains :
+ * - The board as the root node
+ * - A number of chess pieces depending on the game state
+ *
+ * @param context The context used to load the 3d models
+ * @param lifecycleScope A scope that is used to launch the model loading
+ * @param board The board that contains the displayed game state
+ */
+class ChessScene(
+    private val context: Context,
+    private val lifecycleScope: LifecycleCoroutineScope,
```

Comment:
Question : is a `LifecyleCoroutineScope` needed, or would a `CoroutineScope` be enough ?

Reply:
The method used to load 3D models request a `LifecyleCoroutineScope`